packages/dds/task-manager/src/taskManager.ts

```diff
@@ -222,14 +224,21 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
                     // Delete the pending, because we no longer have an outstanding op
                     this.latestPendingOps.delete(taskId);
                 }
+
+                // Remove complete op from this.pendingCompletedTasks
+                const pendingIds = this.pendingCompletedTasks.get(taskId);
+                assert(pendingIds !== undefined && pendingIds.length > 0, "pendingIds is empty");
+                if (pendingIds.length === 1) {
+                    this.pendingCompletedTasks.delete(taskId);
+                } else {
+                    const index = pendingIds.indexOf(pendingOp.messageId);
```

Comment:
Since we should be adding/removing these in order, it should be safe to pendingIds.shift() and assert if it's not the one we expect (basically like https://github.com/microsoft/FluidFramework/blob/main/packages/dds/map/src/mapKernel.ts#L615)

Ideally validate the ID even in the case `pendingIds.length === 1`